externals/CMakeLists.txt

```
@@ -14,6 +14,9 @@ set(BUILD_SHARED_LIBS OFF)
 # Skip install rules for all externals
 set_directory_properties(PROPERTIES EXCLUDE_FROM_ALL ON)
 
+# Robin-map
+add_subdirectory(robin-map)
```

Could you please add `find_package(tsl-robin-map CONFIG)` to the top-level CMakeLists.txt and change this to the following:

```suggestion
if (NOT TARGET tsl::robin_map)
    add_subdirectory(robin-map)
endif()
```

Also, you don't seem to use the `tsl::robin_map` target anywhere in the CMake files, you should probably add `target_link_libraries(<target-name> PRIVATE tsl::robin_map)` in the relevant place(s).